src/pke/lib/cryptocontext.cpp

```diff
@@ -788,30 +788,31 @@ std::unordered_map<uint32_t, DCRTPoly> CryptoContextImpl<DCRTPoly>::ShareKeys(co
                 for (size_t k = 0; k < vecSize; k++) {
                     auto modq_k = elementParams->GetParams()[k]->GetModulus();
 
-                    NativeVector powtempvec(ring_dimension, modq_k);
                     NativePoly powtemppoly(elementParams->GetParams()[k], Format::COEFFICIENT);
                     NativePoly fevalpoly(elementParams->GetParams()[k], Format::COEFFICIENT, true);
 
                     NativeInteger powtemp(1);
                     for (size_t t = 1; t < threshold; t++) {
                         powtemp = powtemp.ModMul(i, modq_k);
 
-                        for (size_t d = 0; d < ring_dimension; d++) {
-                            powtempvec.at(d) = powtemp;
+                        NativeVector powtempvec(ring_dimension, modq_k);
+                        for (size_t i = 0; i < powtempvec.GetLength(); ++i) {
+                            // TODO (dsuponit): should we have a contructor to get a value for all m_data elements in NativeVector
```

Comment:
Change the comment to "... should we have a constructor to set all m_data elements in NativeVector to the same value?"

Reply:
The constructor already exists
NativeVectorT(usint length, const IntegerType& modulus, const IntegerType& val);

Reply:
That constructor sets all m_data elements to val.Mod(modulus), but we need to get them assigned to val

Reply:
It is supposed to be val.Mod(modulus). Equivalent to powtemp.Mod(modq_k).